File: UtilityMethods.m

classdef UtilityMethods
    methods (Static)
        function p = lin_prctile(data,q)
            n=numel(data);
            if n==1
                p=data(1);
            else
                data=sort(data);
                x=(0:(numel(data)-1))/(numel(data)-1)*100; % x-axis
                p=interp1(x,data,q);
            end
        end
        function p2p=peak2peak(x)
            p2p=max(x)-min(x);
        end
    end
end

File: utility_methods.py
"""Utility methods translated from MATLAB."""
import numpy as np
from scipy.interpolate import interp1d


class UtilityMethods:
    @staticmethod
    def lin_prctile(data: np.ndarray, q: float) -> float:
        """Linear percentile interpolation matching MATLAB logic."""
        n = len(data)
        if n == 1:
            return float(data[0])
        else:
            data = np.sort(data)
            x = np.linspace(0, 100, n)
            p = interp1d(x, data, kind='linear', fill_value='extrapolate')(q)
            return float(p)

    @staticmethod
    def peak2peak(x: np.ndarray) -> float:
        """Return peak-to-peak (max - min)."""
        return float(np.max(x) - np.min(x))
